File: demo2/src/main/java/org/mosquito/demo2/DemoRunner.java
package org.mosquito.demo2;

import org.mosquito.demo2.dto.OrderDto;
import org.mosquito.demo2.dto.OrderExtDto;
import org.mosquito.demo2.service.IOrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class DemoRunner implements CommandLineRunner {

    @Autowired
    private IOrderService orderService;

    @Override
    public void run(String... args) throws Exception {
        listAllOrderWithItems();
    }

    private void listAllOrders() {
        OrderDto cond = new OrderDto();
        List<OrderDto> list = orderService.selectAll(cond);
        System.out.println(list);
    }

    private void listAllOrderWithItems() {
        OrderDto dto = new OrderDto();
        List<OrderExtDto> list = orderService.findListExt(dto);
        System.out.println(list);
    }

}

File: demo3/src/main/scala/org/mosquito/demo3/DemoRunner.scala
package org.mosquito.demo3

import org.mosquito.demo3.dto.{OrderDto, OrderExtDto}
import org.mosquito.demo3.service.IOrderService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class DemoRunner extends CommandLineRunner {

  @Autowired
  private val orderService: IOrderService = null

  override def run(args: String*): Unit = {
    println("-------- scala demo runner begin -------")
//    listAllOrders()
//    listAllOrderWithItems()
//    listAllOrderWithItems2()
    test()

    println("-------- scala demo runner endx -------")
  }

  def listAllOrders(): Unit = {
    println(orderService.selectAll2(new OrderDto))
  }

  def listAllOrderWithItems(): Unit = {
    println(orderService.findListExt(new OrderDto))
  }

  def listAllOrderWithItems2(): Unit = {
    println(orderService.findListExt2(new OrderDto))
  }

  def test(): Unit = {
    println(orderService.findListScala(new OrderDto))
  }
}
